fees/incrementor_test.go

```diff
@@ -0,0 +1,216 @@
+/* Mysterium network payment library.
+ *
+ * Copyright (C) 2021 BlockDev AG
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package fees
+
+import (
+	"context"
+	"math/big"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGasPriceIncrementor(t *testing.T) {
+	chid := int64(1)
+	t.Run("gas price is increased and tx is successfull", func(t *testing.T) {
+		originalGasPrice := big.NewInt(1)
+		org := types.NewTransaction(1, common.HexToAddress("0x1"), big.NewInt(1), 1, originalGasPrice, []byte{})
+		opts := defaultOpts()
+		st := &mockStorage{}
+		c := newClient(big.NewInt(2))
+
+		sg := signer{}
+		inc := NewGasPriceIncremenetor(time.Second, st, map[int64]Client{chid: c}, sg.SignatureFunc)
+		go inc.Run()
```

Comment:
defer the stop here.

Reply:
I dont defer on purpose, to avoid accidental race conditions where I'm still pulling data eventhough I should be done already.

I guess we could try and work around that somehow.